zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1003,6 +1003,45 @@ public static List<ACL> fixupACL(String path, List<Id> authInfo, List<ACL> acls)
             throw new KeeperException.InvalidACLException(path);
         }
         List<ACL> rv = new ArrayList<>();
+
+        // Overwrite the acl list for users (except super user) when znode group acl feature is on;
+        // Set znode ACL to the corresponding znode group acl designated ACL instead.
```

Comment:
"Set x509 ZNode ACL as the client's corresponding domain name. This domain name denotes a ZNode group the client belongs to and can be derived from the client URI-domain mapping (UriDomainMappingHelper)."